File: easyEventBus/src/main/java/com/wly/easyeventbus/annotation/Subscriber.java
package com.wly.easyeventbus.annotation;

import com.wly.easyeventbus.ThreadModel;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * author: wangliyun
 * date: 2020/5/26
 * description:
 */
@Target(ElementType.METHOD)
@Retention(RetentionPolicy.RUNTIME)
public @interface Subscriber {

    ThreadModel threadMode() default ThreadModel.POSTING;
}

File: easyEventBus/src/main/java/com/wly/easyeventbus/annotation/Subscriber.kt
package com.wly.easyeventbus.annotation

import com.wly.easyeventbus.ThreadModel
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * author: wangliyun
 * date: 2020/5/26
 * description:
 */
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(RetentionPolicy.RUNTIME)
annotation class Subscriber(val threadMode: ThreadModel = ThreadModel.POSTING)
